src/api/MetadataQuery.js

```diff
@@ -5,26 +5,51 @@
  */
 
 import getProp from 'lodash/get';
-import omit from 'lodash/omit';
+import find from 'lodash/find';
 import Base from './Base';
+import MetadataAPI from './Metadata';
 import { CACHE_PREFIX_METADATA_QUERY, ERROR_CODE_METADATA_QUERY } from '../constants';
 import { ITEM_TYPE_FILE } from '../common/constants';
 import type {
     MetadataQuery as MetadataQueryType,
     FlattenedMetadataQueryResponse,
     FlattenedMetadataQueryResponseEntry,
     FlattenedMetadataQueryResponseEntryMetadata,
+    FlattenedMetadataQueryResponseEntryMetadataField,
     MetadataQueryResponse,
     MetadataQueryResponseEntry,
     MetadataQueryResponseEntryMetadata,
 } from '../common/types/metadataQueries';
 
+const FIELD_TYPE_MULTISELECT: 'multiSelect' = 'multiSelect';
+const FIELD_TYPE_ENUM: 'enum' = 'enum';
+
 class MetadataQuery extends Base {
     /**
      * @property {string}
      */
     key: string;
 
+    /**
+     * @property {string}
+     */
+    templateKey: string;
+
+    /**
+     * @property {string}
+     */
+    templateScope: string;
+
+    /**
+     * @property {object}
+     */
+    templateInfo: MetadataTemplate;
+
+    /**
+     * @property {object}
+     */
+    metadataQueryResponse: MetadataQueryResponse;
+
```

Comment:
Can we add property descriptions? Or just remove it since the flow type somewhat describes it.